client/src/main/scala/org/http4s/client/PoolManager.scala

```diff
@@ -90,29 +88,33 @@ private final class PoolManager[F[_], A <: Connection[F]](
     * @param key      The RequestKey for the Connection.
     * @param callback The callback to complete with the NextConnection.
     */
-  private def createConnection(key: RequestKey, callback: Callback[NextConnection]): Unit =
+  private def createConnection(key: RequestKey, callback: Callback[NextConnection]): F[Unit] =
     if (numConnectionsCheckHolds(key)) {
-      incrConnection(key)
-      unsafeRunAsync(builder(key)) {
-        case Right(conn) =>
-          IO(callback(Right(NextConnection(conn, fresh = true))))
-        case Left(error) =>
-          disposeConnection(key, None)
-          IO(callback(Left(error)))
-      }
+      incrConnection(key) *>
+        F.liftIO {
+          F.runAsync(Async.shift(executionContext) *> builder(key)) {
```

Comment:
This doesn't seem necessary to do at all considering we're in `F[_]` now.

This should be done without runAsync and `liftIO` imo.